datasource/ethereum/src/block_stream.rs

```diff
@@ -401,145 +306,107 @@ where
-                                                    to_block_hash_opt
-                                                        .ok_or_else(|| {
-                                                            format_err!("Ethereum node could not find block with number {}", to)
-                                                        })
-                                                        .map(|to_block_hash| {
-                                                            ReconciliationStep::AdvanceToDescendantBlock {
-                                                                from: subgraph_ptr,
-                                                                to: (to_block_hash, to).into(),
-                                                            }
-                                                        })
-                                                })
-                                        )
-                                    } else {
-                                        // The next few interesting blocks are at descendant_ptrs.
-                                        // In particular, descendant_ptrs is a list of all blocks
-                                        // between subgraph_ptr and descendant_ptrs.last() that
-                                        // contain relevant events.
-                                        // This will allow us to advance the subgraph_ptr to
-                                        // descendant_ptrs.last() while being confident that we did
-                                        // not miss any relevant events.
-
-                                        // Load the blocks
+                            let logger = ctx.logger.clone();
+                            debug!(ctx.logger, "Scanning blocks [{}, {}]", from, to);
+                            Box::new(
+                                ctx.eth_adapter
+                                    .blocks_with_triggers(
+                                        ctx.logger.clone(),
+                                        ctx.chain_store.clone(),
+                                        ctx.metrics.ethrpc_metrics.clone(),
+                                        from,
+                                        to,
+                                        log_filter.clone(),
+                                        call_filter.clone(),
+                                        block_filter.clone(),
+                                    )
+                                    .map(move |descendant_blocks| {
                                         debug!(
-                                            ctx.logger,
-                                            "Found {} relevant block(s)",
-                                            descendant_ptrs.len()
+                                            logger,
+                                            "Found {} trigger(s)",
```

Comment:
I think it's useful here to log the number of blocks, don't we log triggers farther down the pipeline?

Reply:
You're right, seems I got confused here. I've moved this log into `triggers_in_block` and it now logs the number of blocks again.